Marlin/src/gcode/motion/G2_G3.cpp

```diff
@@ -70,6 +82,7 @@ void plan_arc(
   float r_P = -offset[0], r_Q = -offset[1];
 
   const float radius = HYPOT(r_P, r_Q),
+              start_z  = current_position[Z_AXIS],
```

Comment:
Since arcs can now be drawn vertically and not just in the XY plane, this should be using the L (linear) axis, not Z.
```cpp
  start_L = current_position[L_axis],
```

Reply:
OK!    Do you agree the use of start_L should be applied in the case of Bi-Linear also?   Because the leveling is only applied, and never un-applied, I think Bi-Linear will see the nozzle move too much also.

Reply:
> But, is there any possibility of integrating UBL with the planner so no workaround is required? 

Yes.   And in fact...   If you look at the original commented out code I was going to fix it by shifting the planner:: call to  ubl.line_to_destination_cartesian(MMS_SCALED(feedrate_mm_s), active_extruder);

But instead...  I decided to take a baby step and correct the problem with original planner:: code.

The big reason ubl_motion.cpp is a separate file is because it is more efficient from a CPU cycle count.   It optimizes a number of special cases to reduce computation over head.  It recognizes:
- movement within the same mesh row
- movement within the same mesh column
- more efficient diagonal movements across mesh lines.

But this comes at the cost of  more code that is harder to understand.     Any way...  The reason for this explanation is to suggest doing the opposite of what you said.   Perhaps it makes sense to move ubl's motion routine down into the planner so everybody can use the more faster code?

Reply:
As I vaguely recall, something in `ubl_motion` needs to be kicked off so it knows it needs to recalculate for a new XY position, and if positions are changed outside of its purview then it loses track… or something like that. Is that the general gist?

Reply:
Ah...   I'm not sure about that.   That may have been true before all the encapsulation work was done.   But right now...   I'm not aware  of an issue like.